modyn/supervisor/internal/pipeline_executor/evaluation_executor.py

```diff
@@ -355,9 +355,6 @@ def get_failure_reason(eval_aborted_reason: EvaluationAbortedReason) -> str:
         eval_data = self.grpc.get_evaluation_results(response.evaluation_id)
         self.grpc.cleanup_evaluations([response.evaluation_id])
 
-        assert len(eval_data) == len(intervals), f"We expected {len(intervals)} intervals, but got {len(eval_data)}."
-        assert eval_data
```

Comment:
when there is an empty dataset, `len(eval_data)` will be less than `len(intervals)` by 1